### PR TITLE
fix various linting issues  in preparation of golangci-lint update

### PR DESCRIPTION
**Pull request by thaJeztah**
See individual commits for details

- relates to https://github.com/moby/moby/pull/44089

### CODE REVIEW

File: libnetwork/networkdb/cluster.go [hunk 1]
```diff
@@ -718,7 +718,7 @@ func randomOffset(n int) int {
 		return 0
 	}
 
-	val, err := rand.Int(rand.Reader, big.NewInt(int64(n)))
+	val, err := rand.Int(rand.Reader, big.NewInt(int64(n))) // #nosec G404 -- False positive; see https://github.com/securego/gosec/issues/862
```
Comment (thaJeztah) [inline libnetwork/networkdb/cluster.go:721]: Opened a ticket and a pull request to fix the linter;

- https://github.com/securego/gosec/issues/862
- https://github.com/securego/gosec/pull/863

**Review by AkihiroSuda — APPROVED** [review]

File: libnetwork/networkdb/cluster.go (hunk 1 above)
Comment (samuelkarp) [reply libnetwork/networkdb/cluster.go:721]: This is merged now; should we get the new version of gosec rather than marking it nolint?

Comment (thaJeztah) [reply libnetwork/networkdb/cluster.go:721]: I was eyeing the repository, but it's not yet in a release, and after it's been released also needs to make its way into golangci-lint.

I can add a reminder comment in https://github.com/moby/moby/pull/44089 to remove this "nolint" for when we update it to a version that has this fix; would that work?

Comment (samuelkarp) [reply libnetwork/networkdb/cluster.go:721]: Yep, that'd be fine.

**Review by samuelkarp — APPROVED** [review]

**Pull request merged**